nifi-nar-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/ParseNetflowv5.java

```diff
@@ -0,0 +1,258 @@
+                break;
+            case DESTINATION_CONTENT:
+                generateJSON(multipleRecords, session, flowFile, parser, processedRecord, buffer);
+                break;
+            }
+            // Create a provenance event recording the routing to success
+            multipleRecords.forEach(recordFlowFile -> session.getProvenanceReporter().route(recordFlowFile, REL_SUCCESS));
+            session.getProvenanceReporter().route(flowFile, REL_ORIGINAL);
+            // Ready to transfer and commit
+            session.transfer(flowFile, REL_ORIGINAL);
+            session.transfer(multipleRecords, REL_SUCCESS);
+            session.adjustCounter("Records Processed", processedRecord, false);
+            session.commit();
+        } catch (Exception e) {
+            // The flowfile has failed parsing & validation, routing to failure
+            getLogger().error("Failed to parse {} as a netflowv5 message due to {}; routing to failure", new Object[] { flowFile, e });
+            // Create a provenance event recording the routing to failure
+            session.getProvenanceReporter().route(flowFile, REL_FAILURE);
+            session.transfer(flowFile, REL_FAILURE);
+            session.commit();
+            return;
+        } finally {
+            session.rollback();
+        }
+    }
+
+    private void generateJSON(final List<FlowFile> multipleRecords, final ProcessSession session, final FlowFile flowFile, final Netflowv5Parser parser, final int processedRecord, final byte[] buffer)
+            throws JsonProcessingException {
+        int numberOfRecords = processedRecord;
+        FlowFile recordFlowFile = flowFile;
+        int record = 0;
+        while (numberOfRecords-- > 0) {
+            ObjectNode results = mapper.createObjectNode();
+            // Add Port number and message format
+            results.set("port", mapper.valueToTree(parser.getPortNumber()));
+            results.set("format", mapper.valueToTree("netflowv5"));
+
+            recordFlowFile = session.clone(flowFile);
+            // Add JSON Objects
+            generateJSONUtil(results, parser, record++);
+
+            recordFlowFile = session.write(recordFlowFile, new OutputStreamCallback() {
+                @Override
+                public void process(OutputStream out) throws IOException {
+                    try (OutputStream outputStream = new BufferedOutputStream(out)) {
+                        outputStream.write(mapper.writeValueAsBytes(results));
+                    }
+                }
+            });
+            // Adjust the FlowFile mime.type attribute
+            recordFlowFile = session.putAttribute(recordFlowFile, CoreAttributes.MIME_TYPE.key(), "application/json");
+            // Update the provenance for good measure
+            session.getProvenanceReporter().modifyContent(recordFlowFile, "Replaced content with parsed netflowv5 fields and values");
```

Comment:
If you go with `create` you can get rid of this. Something to keep in mind here is that depending on the volume and velocity of the data, you could easily overpower the older provenance repository. The write-ahead one could probably handle it, but you might cause problems for people who didn't migrate (and I'm not sure if the write-ahead one is default now)